### PR TITLE
Treat failed initializer as error (PR 291 continued)

### PR DESCRIPTION
**Pull request by irumaru**
PullRequest 291 has been out of date, and I believe that more time has passed and merging has become more difficult, so I have reworked the PR.
The changes are basically the same as in PullRequest 291, with a few modifications based on comments.

https://github.com/grafana/k6-operator/pull/291

Thanks to JorTurFer and yorugac.

### CODE REVIEW

**Comment by irumaru** [comment]
Hello.
When do you plan to review this?
@yorugac

File: controllers/common.go [hunk 1]
```diff
@@ -54,7 +55,12 @@ func inspectTestRun(ctx context.Context, log logr.Logger, k6 v1alpha1.TestRunI,
 	}
 
 	// there should be only 1 initializer pod
-	if podList.Items[0].Status.Phase != corev1.PodSucceeded && podList.Items[0].Status.Phase != corev1.PodFailed {
+	if podList.Items[0].Status.Phase == corev1.PodFailed {
+		returnErr = errors.New("init job has failed")
```
Comment (yorugac) [inline controllers/common.go:59]:
```suggestion
		returnErr = errors.New("initalizer job has failed")
```

**Review by yorugac — CHANGES_REQUESTED** [review]
Hi @irumaru, thanks for picking this up :raised_hands: Sorry for the delay.

I think this change is fine. I'll test it a bit in coming days and if nothing comes up, I'll merge it in. Meanwhile, there's a small verbosity request for logs - please see the comment.

**Comment by irumaru** [comment]
Thank you for your request.
This is easier to understand and will be changed!

**Review by yorugac — APPROVED** [review]
LGTM! Thanks again @irumaru.

**Pull request merged**